src/sql/src/pure/error.rs

```diff
@@ -203,3 +203,49 @@ impl LoadGeneratorSourcePurificationError {
         }
     }
 }
+
+/// Logical errors detectable during purification for a KAFKA SOURCE.
+#[derive(Debug, Clone, thiserror::Error)]
+pub enum KafkaSinkPurificationError {
+    #[error("{0} is not a KAFKA CONNECTION")]
+    NotKafkaConnection(FullItemName),
+    #[error("admin client errored")]
+    AdminClientError(String),
+    #[error("zero brokers discovered in metadata request")]
+    ZeroBrokers,
```

Comment:
If you are very careful with thiserror, you can use `Arc<anyhow::Error>` as a useful variant: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=9aedb95c48bb501214f7d6e8fba15b7d